json-rpc/src/transactions.rs

```diff
@@ -0,0 +1,148 @@
+                        script_name: code.clone(),
+                        script_args: UserTranscationArgs {
+                            peer_to_peer: if code == "peer_to_peer" {
+                                if let [TransactionArgument::Address(receiver), TransactionArgument::U64(amount)] =
+                                    &args[..]
+                                {
+                                    Some(PeerToPeerArgs {
+                                        receiver: receiver.to_string(),
+                                        amount: *amount,
+                                    })
+                                } else {
+                                    bail!("Unable to parse peer_to_peer arguments")
+                                }
+                            } else {
+                                None
+                            },
+                        },
+                    }),
+                })
+            }
+        }
+    }
+}
+
+/// Returns transactions by range
+pub(crate) async fn get_transactions(
+    libra_db: Arc<LibraDB>,
+    params: Vec<Value>,
+) -> Result<Vec<TransactionView>> {
+    let p_start_version: Number = serde_json::from_value(params[0].clone())?;
+    let p_limit: Number = serde_json::from_value(params[1].clone())?;
+    let _include_events: bool = serde_json::from_value(params[2].clone())?;
+
+    let start_version = p_start_version
+        .as_u64()
+        .ok_or(format_err!("Invalid start_version"))?;
+    let limit = p_limit.as_u64().ok_or(format_err!("Invalid limit"))?;
+
+    ensure!(
+        limit > 0 && limit <= 1000,
+        "limit must be smaller than 1000"
+    );
+
+    let handler = libra_db.get_backup_handler();
+    let iter = handler
+        .get_transaction_iter(start_version, limit)
+        .expect("Unable to get transaction iterator!");
+
+    let mut result: Vec<TransactionView> = vec![];
+    for (v, tx) in iter.enumerate() {
+        result.push(TransactionView {
+            version: start_version + v as u64,
+            transaction: tx?.try_into()?,
+        });
+    }
+
+    Ok(result)
+}
+
+/// Returns account transaction by account and sequence_number
+pub(crate) async fn get_account_transaction(
+    libra_db: Arc<LibraDB>,
+    params: Vec<Value>,
+) -> Result<Option<TransactionView>> {
+    let p_account: String = serde_json::from_value(params[0].clone())?;
+    let p_sequence: Number = serde_json::from_value(params[1].clone())?;
+    let include_events: bool = serde_json::from_value(params[2].clone())?;
+
+    let account = AccountAddress::try_from(p_account)?;
+    let sequence = p_sequence
+        .as_u64()
+        .ok_or(format_err!("Invalid sequence number"))?;
+
+    if let Some(tx) = libra_db
+        .get_txn_by_account(
+            account,
+            sequence,
+            libra_db.get_latest_version()?,
+            include_events,
+        )
+        .expect("Unable to get transaction!")
```

Comment:
I think we should use `?` instead of expect so the error bubbles up and handled by warp?

Reply:
yes, please try to avoid panicking in prod code

Reply:
yes, please try to avoid panicking in production code

Reply:
fixed.